poh/src/transaction_recorder.rs

```diff
@@ -0,0 +1,144 @@
+use {
+    crate::poh_recorder::{PohRecorderError, Record, Result},
```

Comment:
some circular dependencies in the crate here - we could move these to their own modules that both `transaction` and `poh` recorder use. the types are tiny and I don't have a strong preference for either way